astropy/units/quantity.py

```diff
@@ -1375,9 +1380,12 @@ def complex_formatter(value):
 
         # Format unit
         # [1:-1] strips the '$' on either side needed for math mode
-        latex_unit = (self.unit._repr_latex_()[1:-1]  # note this is unicode
-                      if self.unit is not None
-                      else _UNIT_NOT_INITIALISED)
+        if self.unit is None:
+            latex_unit = _UNIT_NOT_INITIALISED
```

Comment:
I wanted to write a test for this code path but I am unable to actually trigger it. Can `Quantity.unit` ever be `None` or should these two lines be removed?

Reply:
I've added a test for this case, so the test coverage of the patch should now be 100%. Is this good to merge or should I rebase it to fix the image tests?